api/src/tarball.rs

```diff
@@ -517,6 +523,9 @@ pub enum PublishError {
     column: usize,
   },
 
+  #[error("CommonJS is not allowed {specifier}")]
+  BannedCommonJsExtension { specifier: String },
```

Comment:
I think we can just re-use `PublishError::CommonJs` for this instead of adding a new error code.